source/fundamentals/logging.txt

```diff
@@ -0,0 +1,137 @@
+         an ``ILoggerFactory`` object by using the ``LoggerFactory.Create()`` method.
+       |
+       | **Data Type**: `ILoggerFactory <https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.iloggerfactory?view=dotnet-plat-ext-7.0>`__
+       | **Default**: null
+
+   * - ``MaxDocumentSize``
+     - | Optional. The maximum number of characters for extended JSON documents in logged 
+         messages.
+       |
+       | For example, when the driver logs the ``CommandStarted`` message, it truncates 
+         the ``Command`` field to the specified character limit.
+       |
+       | **Data Type**: {+int-data-type+}
+       | **Default**: 1000
+
+The following code sample shows how to create a ``MongoClient`` that 
+logs all debug messages to the console: 
+
+.. code-block:: csharp 
+    
+   using var loggerFactory = LoggerFactory.Create(b =>
+   {
+     b.AddSimpleConsole();
+     b.SetMinimumLevel(LogLevel.Debug);
+   });
+ 
+   var settings = MongoClientSettings.FromConnectionString("<your connection string>");
+   settings.LoggingSettings = new LoggingSettings(loggerFactory);
+   var client = new MongoClient(settings);
+
+Log Messages by Category
+------------------------
+
+Each message generated by a MongoDB cluster is assigned a *category*. This lets you 
+specify different log levels for different types of messages. 
+
+MongoDB uses the following categories to classify messages:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 25 75
+
+   * - Category
+     - Description
+
+   * - ``MongoDB.Command``
+     - | The progress of commands run against your cluster, including the 
+         ``CommandStarted``, ``CommandSucceeded``, and ``CommandFailed`` events
+
+   * - ``MongoDB.SDAM``
+     - | Changes in the topology of the cluster, including the ``ClusterOpening``, 
+         ``ClusterOpened``, ``ClusterAddingServer``, ``ClusterAddedServer``, and 
+         ``ClusterRemovingServer`` events 
+
+   * - ``MongoDB.ServerSelection``
+     - | The decisions that determine which server to send a particular command to
+
+   * - ``MongoDB.Connection``
+     - | Changes in the cluster connections and sockets 
+
+   * - ``MongoDB.Internal.*``
+     - | Prefix for all other {+driver-short+} internal components
+
+.. tip::
+ 
+   To log messages from all components, use the ``Default`` category. 
+
+To configure the log level of each message category, create a ``Dictionary<string, string>`` 
+where the key is ``"LogLevel:<category>" and the value is the minimum log level of 
+messages in that category. Add the dictionary to a ``ConfigurationBuilder`` object, 
+then add the ``ConfigurationBuilder`` to your ``LoggerFactory``. 
+
+The following code sample shows how to configure a ``MongoClient`` to log two types 
+of messages: 
+
+- All messages with log level ``Error`` or higher from all categories 
+- All messages with log level ``Debug`` or higher from the SDAM category 
```

Comment:
[nit] Since each item in this list starts with a capital letter, the end of each individual list item should have a period.

Reply:
My read of the [style guide](https://www.mongodb.com/docs/meta/style-guide/style/lists/) is that you should only include periods if one or more of the list entries is a full sentence. Did I miss something?

Reply:
Huh. Fair point! I was referencing [AP style guide](https://www.instructionalsolutions.com/blog/bulleted-list-punctuation), not ours. I agree with you after reading through our style guide. My apologies!

Reply:
Good to know! Thanks.